models/classes/class.QtiTestService.php

```diff
@@ -873,19 +875,46 @@ private function deleteTestsFromClassByLabel(
         core_kernel_classes_Class $itemClass
     ): void {
         $testService = $this->getTestService();
-        $itemTreeService = $this->getItemTreeService();
+
+        $deletedTests = [];
+        $deletedItemClasses = [];
+        $refs = [];
 
         foreach ($testClass->getInstances() as $testInstance) {
             if ($testInstance->getLabel() === $testLabel) {
+                $deletedTests[] = $testInstance->getUri();
                 $testService->deleteTest($testInstance);
             }
         }
 
         foreach ($itemClass->getSubClasses() as $subClass) {
             if ($subClass->getLabel() === $itemsClassLabel) {
-                $itemTreeService->deleteClass($subClass);
+                $deletedItemClasses[] = $subClass->getUri();
+                $refs = array_merge($refs, $this->deleteItemClass($subClass));
             }
         }
+
+        $this->getEventManager()->trigger(
+            new QtiTestDeletedEvent($deletedTests, $deletedItemClasses, $refs)
+        );
+    }
+
+    private function deleteItemClass(core_kernel_classes_Class $subClass): array
```

Comment:
This method is not only deleting the item class, but collecting list of resources references. Would not be better to keep the previous implementation and create a method called 'getResourceReferences':

```
    private function getResourceReferences(core_kernel_classes_Resource $item): array
    {
        $qtiItem = $this->getQtiItemService()->getDataItemByRdfItem($rdfItem);
        
        return $this->getElementReferencesExtractor()->extractAll($qtiItem)->getAllReferences();
    }
```

And then you can call this method here https://github.com/oat-sa/extension-tao-testqti/pull/2393/files#diff-05b1f87fb7a793e119a1ba8079639c0bfa12edb96f8e45615c2807361951b09bR893

Isn't' it cleaner and more SOLID? Wdyt?

Reply:
Changed: fc7194217

Reply:
Thanks!